app/dockerdwrapperwithcompose.c

```diff
@@ -241,6 +248,35 @@ setup_sdcard(void)
   return res == 0;
 }
 
+/**
+ * @brief Attempt to verify ownership for path with timeout.
+ *
+ * @param path pointer path to verify.
+ * @param uid the user id to verify.
+ * @param gid the group id to verify.
+ * @param secs the maximum time to wait for verification.
+ * @return true if verified, false otherwise
+ */
+static bool
+verify_ownership(char *path, uid_t uid, uid_t gid, uint secs)
+{
+  struct stat path_stat;
+  uint i = 0;
+
+  while (i++ < secs) {
```

Comment:
could this be handled by without a while-sleep? eg. g_timeout_add_seconds()

Reply:
Every time we loop with a sleep like this, parhand risks being blocked, so we really shouldn't do it.

Yes, this should be done using g_timeout_add_seconds(), together with app_state (which I have introduced on the main branch), so that a callback can do the chmod() call and control if it needs to be called again through the return value.